mobile/src/notifications/notification-listener.ts

```diff
@@ -0,0 +1,68 @@
+/******************************************************************************
+ * Copyright (c) Arcadia, Inc. All rights reserved.
+ ******************************************************************************/
+
+import { ActionsObservable, combineEpics, ofType, StateObservable } from 'redux-observable';
+import { AuthActionType, UserLoggedIn } from '../reducers/auth/auth.action';
+import { catchError, ignoreElements, map, switchMap } from 'rxjs/operators';
+import Push from 'appcenter-push';
+import { EMPTY, from, of, Subject } from 'rxjs';
+import { Action } from 'redux';
+import { openProfile } from '../navigation/navigation.actions';
+import { AppState, DependenciesContainer } from '../reducers/app.reducer';
+import AppCenter from 'appcenter';
+import { installIdReceived, NotificationAction, NotificationActionType } from './notification.actions';
+
+//----------------------------------------------------------------------------
+const notificationsHandler$ = (action$: ActionsObservable<UserLoggedIn>) =>
+    action$.ofType(AuthActionType.userLoggedIn).pipe(
+        switchMap(() => {
+            const notification$ = new Subject<Action>();
+
+            Push.setListener({
+                onPushNotificationReceived: notification => {
+                    if (!notification.customProperties) {
+                        return;
+                    }
+
+                    notification$.next(openProfile());
+                },
+            });
+
+            return notification$;
+        }),
+    );
+
+//----------------------------------------------------------------------------
+const getInstallId$ = (action$: ActionsObservable<UserLoggedIn>) => action$.pipe(
+    ofType(AuthActionType.userLoggedIn),
+    switchMap(() => {
+        return from(AppCenter.getInstallId());
+    }),
+    map(installId => installIdReceived(installId)),
+);
+
+//----------------------------------------------------------------------------
+const notificationsRegister$ = (action$: ActionsObservable<NotificationAction>, _: StateObservable<AppState>, deps: DependenciesContainer) => action$.pipe(
+    ofType(NotificationActionType.installIdReceived),
+    switchMap(action => {
+        return deps.apiClient.put(`/push/device/${action.installId}`);
```

Comment:
This will be `POST` with install id in body.

Reply:
9aef55b1bc5dec636d8949d4a2c11319e2f551c6